internal/router/agent_routes_test.go

```
@@ -89,3 +91,70 @@ func TestAgentSessionCancelRouteRegistered(t *testing.T) {
+	cases := []struct {
+		name   string
+		method string
+		path   string
+	}{
+		{"agent file upload", http.MethodPost, "/api/v1/agents/canvas-1/upload"},
+		{"agent attachment download", http.MethodGet, "/api/v1/agents/attachments/att-1/download"},
+	}
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			resp := httptest.NewRecorder()
+			req := httptest.NewRequest(tt.method, tt.path, nil)
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)